### ISSUE TITLE
[REQ] automatically retry of all http operations on http 429 with retry-after header

### ISSUE THREAD

**Issue by wreggyl** [OpenAPITools/openapi-generator#10384]
### Is your feature request related to a problem? Please describe.

Our server implement an API limiter with status code 429 and the retry-after header. Currently we have to catch the response outsite of the generated client and retry from applicaition level

[https://datatracker.ietf.org/doc/html/rfc6585#section-4](url)

## Describe the solution you'd like

Would be good if the generated code automatically retry the call after the provided time in the retry-after header like it is does on timeouts

## Describe alternatives you've considered

implement the retry logic outsite of the generated client code or patching the generated code

**Comment by wreggyl** [comment]
my client curretnly is in Python and I think I can fix it directly in the mustache file, but currently I do not have an idea how to provide such a change globally and created this FR also as some ground for discussion. Maybe some disagree to put this kind of logic in the generated clients.

**Comment by wreggyl** [comment]
e.g.: for Python I can already get this working with the following configuration without patching anything
retry_strategy = urllib3.Retry(
            total=3,
            backoff_factor=1,
            status_forcelist=[429],
            method_whitelist=["HEAD", "GET", "OPTIONS", "POST"],
        )
configuration.retries = retry_strategy

**Comment by sandeepkumar-one** [comment]
Please go through this doc : https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html

Try setting `status`, as default is 0.
Try setting `respect_retry_after_header` as true

**Comment by wreggyl** [comment]
@sandeepkumar-one works like a charm, but I had to set status = 1 to retry it at least one time

**Comment by wreggyl** [comment]
the initial problem on my site was that I tested it against the misconfigured server which didn't send the retry-after-header. As urllib3 has retryafter as default we can close the this FR, because everything is more or less working like a charm => only in case there is not retry-after-header but you get a 429 my previous workaround works => maybe this is an idea as a default as well?

**Comment by sandeepkumar-one** [comment]
Yes `status` should be more than 1. Its default is 0. So it won't work if value is 0.

**Comment by raulbocanegra** [comment]
This is still happening because [`RESTClientObject`](https://github.com/OpenAPITools/openapi-generator/blob/7e736453033caeee932979f652326d84e9771896/modules/openapi-generator/src/main/resources/python-prior/rest.mustache#L37) check for a `retries` in its configuration parameter and pass it to a `urllib3.PoolManager`.

When we do a request, the `RESTClientObject` of the api client, calls [urllib3.PoolManager.request](https://urllib3.readthedocs.io/en/latest/reference/urllib3.poolmanager.html#urllib3.PoolManager.request) which after some validations calls [urllib3.PoolManager.urlopen](https://urllib3.readthedocs.io/en/latest/reference/urllib3.poolmanager.html#urllib3.PoolManager.urlopen). It is in this function where the retry logic takes place, but this method expects the [retry object](https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html) to be passed as parameter in the `kw` dict, but the generated code does not call this function with it. Instead, it is storing the retry object inside an instance variable (`connection_pool_kw`) of the `urllib3.PoolManager`, that it is not being used when calling `url_open`.

A solution for this would be that `RESTClientObject` which may have the retry object stored in its `configuration` forwarded it to the `RESTClientObject.request` method setting the paramenter `urlopen_kw` of the `self.pool_manager.request` call if it is present in the `RESTClientObject` configuration.

Another solution would be that urllib3 checked for a `retries` parameter in its `connection_pool_kw` instance variable and forwarded it to `urllib3.PoolManager.urlopen`when doing a `urllib3.PoolManager.request`
